lib/inferno/config/boot/validator.rb

```diff
@@ -7,10 +7,11 @@
     next if Sidekiq.server?
 
     Inferno::Repositories::TestSuites.new.all.each do |suite|
-      suite.fhir_validators.each do |name, validators|
+      suite.fhir_validators.each do |name, validators, required_suite_options|
         validators.each_with_index do |validator, index|
           if validator.is_a? Inferno::DSL::FHIRResourceValidation::Validator
-            Inferno::Jobs.perform(Inferno::Jobs::InvokeValidatorSession, suite.id, name.to_s, index)
+            Inferno::Jobs.perform(Inferno::Jobs::InvokeValidatorSession, suite.id, name.to_s, index,
+                                  required_suite_options)
```

Comment:
It doesn't seem like there is currently any code in core that exercises creating validator sessions with options. Have you tried this with [Dylan's g10 PR](https://github.com/onc-healthit/onc-certification-g10-test-kit/pull/488)? It doesn't work when I try it with this change.

```
13:01:46 worker.1  | E, [2024-02-20T13:01:46.587085 #51795] ERROR -- : SQLite3::SQLException: no such column: us_core_version: SELECT * FROM `validator_sessions` WHERE ((`test_suite_id` = 'g10_certification') AND (`validator_name` = 'default') AND (`suite_options` = (`us_core_version` = 'us_core_4'))) LIMIT 1
13:01:46 worker.1  | 2024-02-20T18:01:46.594Z pid=51795 tid=9gb class=Inferno::Jobs::InvokeValidatorSession jid=55fc3b4eebfae2b253df5e6d elapsed=0.009 INFO: fail
13:01:46 worker.1  | 2024-02-20T18:01:46.594Z pid=51795 tid=9gb WARN: {"context":"Job raised exception","job":{"retry":true,"queue":"default","args":["g10_certification","default",1,null],"class":"Inferno::Jobs::InvokeValidatorSession","jid":"55fc3b4eebfae2b253df5e6d","created_at":1708451870.651393,"enqueued_at":1708452106.580876,"error_message":"SQLite3::SQLException: no such column: us_core_version","error_class":"Sequel::DatabaseError","failed_at":1708451870.684702,"retry_count":3,"retried_at":1708451988.8564491}}
13:01:46 worker.1  | 2024-02-20T18:01:46.594Z pid=51795 tid=9gb WARN: Sequel::DatabaseError: SQLite3::SQLException: no such column: us_core_version
```

Reply:
Based on the query it looks like it's converting the dict to an invalid query structure.
```
SELECT * FROM `validator_sessions` WHERE ((`test_suite_id` = 'g10_certification') AND (`validator_name` = 'default') AND (`suite_options` = (`us_core_version` = 'us_core_4')))
```
Specifically the ```(`suite_options` = (`us_core_version` = 'us_core_4'))``` part.

We may have to turn the options dict into a string before sending it to the database

Reply:
Yes, it's also important to verify that it is being properly serialized/deserialized when being passed to the job.